### PR TITLE
set tracker reminder

### PR DESCRIPTION
**Pull request by wbaccinelli**
client modification for: https://github.com/PerfectFit-project/virtual-coach-issues/issues/170

### CODE REVIEW

File: niceday_client/definitions.py [hunk 1]
```diff
@@ -1,6 +1,9 @@
-from enum import IntEnum
+from enum import IntEnum, Enum
 
 USER_PROFILE_KEYS = ['firstName', 'lastName', 'location', 'birthDate', 'gender']
 
 class Tracker(IntEnum):
     SMOKING = 1
+
+class TrackerName(Enum):
+    SMOKING = 'tracker_smoking'
```
Comment (svenvanderburg) [inline niceday_client/definitions.py]: We could also try and combine this? It's of course super confusing because both are actually identifiers.

File: niceday_client/niceday_client.py [hunk 2]
```diff
@@ -186,3 +187,47 @@ def get_smoking_tracker(self, user_id: int, start_time: datetime.datetime,
...
         response_json = json.loads(query_response.content)
         return response_json
+
+    def set_tracker_reminder(self, user_id: int, tracker_name: str, reminder_title: str, recurrence_rule: rrule):
+        """
+        Set tracker reminder for a specific user.
+
+        Example Usage:
+            ```
+            client.set_tracker_reminder(12345, TrackerName.SMOKING.value, "This is a tracker", rrule(DAILY,dtstart=datetime.datetime(2022, 5, 12, 0, 0),until=datetime.datetime(2022, 5, 13, 0, 0)))
```
Comment (svenvanderburg) [inline niceday_client/niceday_client.py]: This line is too long to read easily.

File: niceday_client/niceday_client.py [hunk 3]
```diff
@@ -186,3 +187,47 @@ def get_smoking_tracker(self, user_id: int, start_time: datetime.datetime,
...
+
+        Args:
+            user_id: ID of the user we want to set tracker statuses for
+            tracker_name: Name of the tracker to set the reminder for (see
+            https://github.com/senseobservationsystems/goalie-js/issues/840
+            on how to get tracker name, for example: smoking tracker has name=tracker_smoking).
+            Use the TrackerName enum defined in definitions.py instead of hardcoding
+            string values. e.g. 'TrackerName.SMOKING' may be used instead of 'tracker_smoking'.
```
Comment (svenvanderburg) [inline niceday_client/niceday_client.py]:
```suggestion
            tracker_name: Name of the tracker to set the reminder for (see
                https://github.com/senseobservationsystems/goalie-js/issues/840
                on how to get tracker name, for example: smoking tracker has name=tracker_smoking).
                Use the TrackerName enum defined in definitions.py instead of hardcoding
                string values. e.g. 'TrackerName.SMOKING' may be used instead of 'tracker_smoking'.
```

**Review by svenvanderburg — APPROVED** [review]
Nice additions. I left some comments. Also: did you think about adding a test? Especially if you already wrote some code to test, it would be good to include it. (we have some tests marked as integration_test that we do not run automatically, but are handy for development).

**Comment by wbaccinelli** [comment]
The problem with the testing here, is that it will actually set a reminder that has to be removed before setting another one (is not possible so set the same type of reminder in overlapping times). This either requires another functionality to be implemented, or the tests to be executed successfully only once in a hour (one hour seems to be the minimum duration of a reminder, after having performed some trials). 
The other possibility is mocking the response, but as this should be an integration test, I'm wondering whether this makes sense.
